php/WP_CLI/Dispatcher/CommandFactory.php

```diff
@@ -110,5 +128,67 @@ private static function create_composite_command( $parent, $name, $callable ) {
 	private static function is_good_method( $method ) {
 		return $method->isPublic() && !$method->isStatic() && 0 !== strpos( $method->getName(), '__' );
 	}
-}
 
+	/**
+	 * Gets the document comment. Caters for PHP directive `opcache.save comments` being disabled.
+	 *
+	 * @param ReflectionMethod|ReflectionClass|ReflectionFunction $reflection Reflection instance.
+	 * @return string|false|null Doc comment string if any, false if none (same as `Reflection*::getDocComment()`), null if error.
+	 */
+	private static function get_doc_comment( $reflection ) {
+		$doc_comment = $reflection->getDocComment();
+
+		if ( false !== $doc_comment || ! ( ini_get( 'opcache.enable_cli' ) && ! ini_get( 'opcache.save_comments' ) ) ) {
+			// Either have doc comment, or no doc comment and save comments enabled - standard situation.
+			if ( ! getenv( 'WP_CLI_TEST_GET_DOC_COMMENT' ) ) {
+				return $doc_comment;
+			}
+		}
+
+		$filename = $reflection->getFileName();
+
+		if ( isset( self::$file_contents[ $filename ] ) ) {
+			$contents = self::$file_contents[ $filename ];
+		} elseif ( is_readable( $filename ) && ( $contents = file_get_contents( $filename ) ) ) {
+			self::$file_contents[ $filename ] = $contents = explode( "\n", $contents );
```

Comment:
I'm thinking though as long as a "\n" is in the EOL, then this is safer - unless the files are Mac formatted circa 1990s with a single "\r"! If it is Windows "\r\n" then the existing code is agnostic to that, so works regardless of file format.